### PR TITLE
fix: move @types/i18next-fs-backend dep to dev-deps

### PR DESCRIPTION
**Pull request by belgattitude**
## Purpose

Avoid having "@types/i18next-fs-backend"  as a dep. 

**tldr;** if latest`@types/i18next-fs-backend"` (> 1.1.0), it will install i18next v21... we'll have both versions hoisted. I suspect packages managers (yarn/npm/pnpm), webpack (4/5), node resolutions (ssr) to give different results leading to bugs like https://github.com/isaachinman/next-i18next/issues/1443

## Possible explanation

Why the `@types/i18next-fs-backend": "^1.0.0"` is problematic here.

- 1.0.0 depends on i18next 19 (that will be installed)
- 1.1.0 (no deps)
- 1.1.2 depends on i18next 21 (that will be installed)

*PS: There's not yet official types for i18next-fs-backend, see https://github.com/i18next/i18next-fs-backend/issues/20*

I'm not sure about the move to devDeps, I could have locked it to `1.0.1` but I haven't seen a re-export of it. So it should be fine. (@isaachinman can you confirm ?, Otherwise we could lock the version or move it to optional peer-deps. If some projects relies on those types, they might have to explicitly add them, not technically a BC but who knows ?)

I've set version to `~1.0.0` in dev-deps... 

When next-18next supports i18next v21, it might need to be updated to `^1.1.0`.

## Possible fixes.

Will probably fix this https://github.com/isaachinman/next-i18next/issues/1443 and hopefully this https://github.com/i18next/react-i18next/issues/1379

I could not give a reproduction example, but here's the explanation here: https://github.com/belgattitude/nextjs-monorepo-example/pull/510 (I could reproduce it partially, but on prod I'm using a different version of yarn so I'm not sure, my workaround was to add i18next: ^20 explicitly).

## Extra

I realized that the example was loading i18next 19 too... Changed the lock: https://github.com/isaachinman/next-i18next/pull/1467/commits/b3c55979a3e27f61f9868c415b6cf3de87db44a1

### CODE REVIEW

**Comment by belgattitude** [comment]
I can confirm it would close this one too https://github.com/i18next/react-i18next/issues/1379.

In the meantime

```json5
  "resolutions": {
    "@types/i18next-fs-backend": "1.1.0", // this would be anough
    "i18next": "^20.6.1" // or to be safe
  },
```

**Comment by isaachinman** [comment]
You'll need to `yarn install` again.

**Comment by isaachinman** [comment]
I don't understand what you are suggesting. You just need to move the dependency from deps into dev deps, and do `yarn install && yarn run-example`.
